internal/osutils/lockfile/pidlock.go

```diff
@@ -94,6 +95,28 @@ func (pl *PidLock) Close(keepFile ...bool) error {
 	return nil
 }
 
+// WaitForLock will attempt to acquire the lock for the duration given
+func (pl *PidLock) WaitForLock(timeout time.Duration) error {
+	lockedErr := &AlreadyLockedError{}
+	timer := time.NewTimer(timeout)
+	for {
+		_, err := pl.TryLock()
+		if err != nil {
+			if !errs.Matches(err, &lockedErr) {
+				return errs.Wrap(err, "Could not acquire lock")
+			}
+
+			select {
+			case <-timer.C:
+				return err
+			default:
+				time.Sleep(100 * time.Millisecond)
+			}
+		}
+		return nil
+	}
```

Comment:
This is a nice design, but I think the channel can be dropped for a simple time.Time.After call: https://play.golang.org/p/AZ9VI4e4HHv